rust/lancedb/src/remote/db.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 LanceDB Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use async_trait::async_trait;
+use serde::Deserialize;
+
+use crate::connection::{ConnectionInternal, CreateTableBuilder, OpenTableBuilder};
+use crate::error::Result;
+use crate::TableRef;
+
+use super::client::RestfulLanceDbClient;
+
+#[derive(Deserialize)]
+struct ListTablesResponse {
+    tables: Vec<String>,
+}
+
+#[derive(Debug)]
+pub struct RemoteDatabase {
+    client: RestfulLanceDbClient,
+}
+
+impl RemoteDatabase {
+    pub fn try_new(
+        uri: &str,
+        api_key: &str,
+        region: &str,
+        host_override: Option<String>,
+    ) -> Result<Self> {
+        let client = RestfulLanceDbClient::try_new(uri, api_key, region, host_override)?;
+        Ok(Self { client })
+    }
+}
+
+#[async_trait]
+impl ConnectionInternal for RemoteDatabase {
+    async fn table_names(&self) -> Result<Vec<String>> {
+        let rsp = self
+            .client
+            .get("/v1/table/")
+            .query(&[("limit", 10)])
+            .query(&[("page_token", "")])
```

Comment:
I guess pagination is TODO?

Reply:
Yeah.  It wasn't part of the `Table` abstract class, so it wasn't documented at all and there was no implementation for the local table.  I figure I can add it back in, with support for both tables, at a later date.  I'll make a PR.

Reply:
I was just referring to the fact we only request the first page here (first 10).

Reply:
Yeah, that's pretty broken.  Still, I think I'd rather defer.  I've added this to https://github.com/lancedb/lancedb/issues/926 and I'm starting to track all the cleanup tasks as part of https://github.com/lancedb/lancedb/issues/926